src/utils/generator.js

```diff
@@ -205,7 +206,17 @@ class ModifiedBindingEntry extends GeneratorEntry {
       return;
     }
     invariant(this.modifiedBinding.value === this.newValue);
-    if (this.newValue) residualFunctionBinding.additionalValueSerialized = context.serializeValue(this.newValue);
+    invariant(
+      residualFunctionBinding.serializedValue,
+      "ResidualFunctionBinding must be referentialized before serializing a mutation to it."
+    );
+    let newValue = this.newValue;
+    if (newValue) {
+      let bindingReference = ((residualFunctionBinding.serializedValue: any): BabelNodeLVal);
+      invariant(t.isLVal(bindingReference), "Referentialized values must be LVals");
```

Comment:
I am always looking for a motivating comment when I see an invariant. When can a serializedValue be something other than an LVal? Why can that never happen here?

Reply:
A serialized value can in general be a BabelNodeExpression (e.g. an object literal), but since the value has been referentialized, it must be an LVal.